### PR TITLE
finished implementing Message Feed! (up to step 4)

### PR DESCRIPTION
**Pull request by faesalm**
finished steps 3-4. Also refactored shared logic in getMessages and getAllMessages methods in Datastore.

### CODE REVIEW

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -54,33 +54,58 @@ public void storeMessage(Message message) {
...
+    List<Message> messages = fetchMessages(query);
+    
+    return messages;
+  }
+  
+  /**
+   * Gets messages posted by all users.
+   *
+   * @return a list of messages posted by all users, or an empty list if no user has posted a 
+   *     message. List is sorted by time descending.
```
Comment (Tetheta) [inline src/main/java/com/google/codeu/data/Datastore.java:71]: Formatting is weird here, why is a on its own line? Might just be code review wrapping though

**Review by Tetheta — CHANGES_REQUESTED** [review]
In general looks pretty good, but hard to read between all the stickler errors.

Are you using an autoformatter perhaps? Seems like the indenting is different than our linter is expecting.

Fix those issues and I'll give it another quick pass to make sure it's good to go.

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 1 above)
Comment (faesalm) [reply src/main/java/com/google/codeu/data/Datastore.java:71]: Yeah I am using Eclipse and it is doing funky things by default. I will go fix now

Comment (faesalm) [reply src/main/java/com/google/codeu/data/Datastore.java:71]: What do you mean 'a in its own line' ? I don't see that on my screen .PS formatting is fixed now!

**Review by Tetheta — APPROVED** [review]

**Comment by Tetheta** [comment]
Feel free to merge when you're comfortable.

I'll have to see if there's a better way to check the styling than a bunch of commits on a PR lol. But seems the only way for now.

**Comment by faesalm** [comment]
I found out you can download the Google style guide to Eclipse and it will auto format. I have not tested it yet but it seems like it works. See this: 
https://github.com/HPI-Information-Systems/Metanome/wiki/Installing-the-google-styleguide-settings-in-intellij-and-eclipse

**Pull request merged**